### DISCUSSION TITLE
Where is the list command alias?

### DISCUSSION THREAD

**Discussion by OldSubSailor** [ohmyzsh/ohmyzsh#12504] [Help]
Just as the title says. I had my own alias for la, and ll (in my .zshrc file) and so does ohmyzsh (somewhere). I can't find where they come from, I am guessing the plugins. I don't know which plugin it is in, and I hope that I can simply comment it out;  or how they are called. In fact, the ohmyzsh **la** alias is my **ll** exactly.  The goal is to swap them out. 
If they are a plugin perhaps the name could be changed to something obvious. 
I would appreciate any advice you can provide.

**Comment by carlosala** [comment]
You could add `zstyle :omz:lib:directories aliases no` on your `zshrc` right before sourcing `oh-my-zsh.sh`. It's explained [here](https://github.com/ohmyzsh/ohmyzsh?tab=readme-ov-file#skip-aliases).

**Reply by OldSubSailor** [reply]
That still does not answer my question as to where they are located. I saw nothing in the library.  Further, that is not clear as to me and/or perhaps to other users that we would be skipping one or all the aliases at one shot e.g. "# Skip **all aliases**, in lib files and enabled plugins" which your link indicates will happen.  So no, it is not explained there, not for just two aliases. 

So far the only thing that I have read that provided any relief was in Customization that reads:
"Initially $ZSH_CUSTOM points to Oh My Zsh's custom directory. Whatever you place inside of it will take precedence over the built-in configuration."  So that means I can write an alias that will override the ohmyzsh commands, but that still does not get me to the commands themselves. I do not now think that they are plugins, so what and where are they, and is a custom alias the only way to over them?

**Reply by carlosala** [reply]
They are located in `lib/directories.zsh`

**Reply by OldSubSailor** [reply]
Thank you for the pointer. Now:

-  How does a new user find them without spending a lot of time searching through each file, if they are aware that the alias is in lib to begin with?
- Perhaps additional instructions along the lines that you pointed out earlier would be of great aid in locating and modifying a few aliases?  The locate DB is dead to us, and which or where _la_ returns a rather useless: _la: aliased to ls -AFG_
- Is writing a new alias file into the Custom directory the best way to override an alias or three, or would editing them in the lib/file be a better way to go?  If so, is there a tool and methodology in place or does one just jump in  with a text editor?

**Reply by carlosala** [reply]
If you search `alias ll` on GitHub search will most likely do the trick.
In order to modify the file, I recommend you copy the same structure in the custom directory. That is, a file in `$ZSH/custom/lib/directories.zsh` with modifications will be loaded instead of the default if it exists.

**Reply by mcornella** [reply]
You can also override a single alias by putting it in your .zshrc file, anywhere after the `source $ZSH/oh-my-zsh.sh`. The last defined alias will be preserved and the previous one overriden.

**Reply by OldSubSailor** [reply]
@mcornella 
Thank you for your reply. I do know that, however, if you take a look at the "new" .zshrc file that is generated and replaced the original users .zshrc,  there is a statement that reads in part "Aliases can be placed here, though oh-my-zsh
 users are encouraged to define aliases within the ZSH_CUSTOM folder.
 For a full list of active aliases, run `alias`."

**Reply by mcornella** [reply]
Oh I see, thanks for spotting that! We must have added it this way to keep the `.zshrc` file as lean as possible.

The files the message is referencing should be immediate children of the custom folder, with `.zsh` extension, such as `$ZSH_CUSTOM/aliases.zsh`.

These files are last to be read (see [^1]), just before loading the theme, so any new changes in there take precedence.

I think we can make the explanation clearer, I'll try to get to it later today.

Thanks again!

[^1]: https://github.com/ohmyzsh/ohmyzsh/blob/0c9b42a863778d67aee22bbd43f296110fe1f047/oh-my-zsh.sh#L212

**Reply by OldSubSailor** [reply]
@carlosala I appreciate that, however, once again I am thinking new user audience analysis and instructions for modification of a few of them. Going to Github to perform a search already complicates things. 

Now let me step back and preface this. I installed ohmyzsh because it is frequently mentioned in a book, _MASTERING THE MACOS TERMINAL: A ZSH GUIDE FOR BEGINNERS TO EXPERTS._    This is a book that I have recommended to BEGINNERS. Clearly, I am no master, however, I have been a technical instructor. 

I personally feel that the ohmyzsh **la** listing is not good for a beginner, whereas mine (alias la="ls -AFG") works for them. The ohmyzsh la is my version of ll which I think that  beginners would call if they wanted to see a long listing.  

You have provided some valuable information and I think that it would be great if whomever writes the ohmyzsh instructions could gather that up, absent going to Github, I think just finding things directly in the lib for example, and perhaps even developing a shortcut to find and edit the desired alias; in a BEGINNERS SECTION.

**Reply by OldSubSailor** [reply]
@mcornella 
Wow, quick response on a Saturday, thank you.
I am not a programmer, but I do believe that even I was able to understand the footnote, thank you. 
Well I wrote just such a file that contains about 5 of my favorite aliases. I tried to explain my concern further in a reply to carlosala.  I had thought about ginning an issue requesting a Document Section for Beginners.  I tried every trick that I know of in an effort to find the aliases on terminal and had no luck although they were sitting right there where carlosala pointed me to.  Find Any File unfortunately would return a list of thousands if one tried searching for "la". :(
